### ISSUE TITLE
[java] False positive ForLoopCanBeForeach

### ISSUE THREAD

**Issue by scottkennedy** [pmd/pmd#1563]
**Affects PMD Version:** 
6.10.0

**Rule:**
ForLoopCanBeForeach

**Description:**
I have code very similar to the following that's producing a `ForLoopCanBeForeach` issue, even though I definitely need the index variable.

**Code Sample demonstrating the issue:**

```java
for (int i = 0; i < mList.size(); i++) {
    mList.get(i).setIndex(i);
}
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
Gradle
